include/nbl/ui/IWindowXCB.h

```diff
@@ -0,0 +1,88 @@
+#ifndef __NBL_I_WINDOW_XCB_H_INCLUDED__
+#define __NBL_I_WINDOW_XCB_H_INCLUDED__
+
+#ifdef _NBL_PLATFORM_LINUX_
+
+#include "nbl/core/util/bitflag.h"
+
+#include "nbl/ui/IWindow.h"
+#include "nbl/ui/XCBConnection.h"
+
+#include <xcb/xproto.h>
+
+namespace nbl::ui
+{
+
+class NBL_API2 IWindowXCB : public IWindow
+{
+    protected:
+        virtual ~IWindowXCB() = default;
+        inline IWindowXCB(SCreationParams&& params) : IWindow(std::move(params)) {}
+
+    public:
+        using IWindow::IWindow;
+
+        const void* getNativeHandle() const { return nullptr; }
+        virtual xcb_window_t getXcbWindow() const = 0;
+        virtual xcb_connection_t* getXcbConnection() const = 0;
```

Comment:
I suggest forward declaring outside the namespace
```cpp
using xcb_window_t = uint32_t;
struct xcb_connection_t;
```
and then following our convention which is
```cpp
struct native_handle_t
{
    xcb_window_t window;
    xcb_connection_t* connection;
};
const native_handle_t& getNativeHandle() const =0;
```